src/io.jl

```diff
@@ -191,22 +159,33 @@ function update_forcing!(model)
         if par in mover_params
             if do_reservoirs
                 for (i, sel_reservoir) in enumerate(sel_reservoirs)
-                    avg = mean(buffer[sel_reservoir])
-                    buffer[sel_reservoir] .= 0
+                    avg = mean(data[sel_reservoir])
+                    data[sel_reservoir] .= 0
                     param_res[par][i] = avg
                 end
             end
             if do_lakes
                 for (i, sel_lake) in enumerate(sel_lakes)
-                    avg = mean(buffer[sel_lake])
-                    buffer[sel_lake] .= 0
+                    avg = mean(data[sel_lake])
+                    data[sel_lake] .= 0
                     param_lake[par][i] = avg
                 end
             end
         end
 
         param_vector = param(model, par)
-        param_vector .= buffer[sel]
+        data_sel = data[sel]
+        if par == (:lateral, :land, :h_riv) || par == (:lateral, :land, :q_riv)
```

Comment:
I don't think we need this check if `h_riv` and `q_riv` are not used as part of the `OverlandFlowSediment` struct. For this to work we probably also need to modify the sediment_config.toml file of the sediment test model.

Reply:
They are not used, but it will error if we attempt to add those missing cells to the `param_vector::Vector{Float64}` in the line below. I was planning to remove them from the `OverlandFlowSediment` and load the forcing into the `RiverSediment` struct directly. When that is ready, we can remove this check.

You're right that if we do this, we probably need to move

```toml
[input.lateral.land]
h_riv = "h"
q_riv = "q"
```

to `[input.lateral.river]`. That would be a breaking change, so perhaps best to still include that in this release?

Reply:
My preference is to resolve this now, we're going anyway from version 0.2.0 to 0.3.0. I think we also need to change tehn the `forcing` part in the toml file to:

```toml
forcing = [
  "vertical.h_land",
  "vertical.interception",
  "vertical.precipitation",
  "vertical.q_land",
  "lateral.river.h_riv",
  "lateral.river.q_riv",
]
```

Reply:
Ok, let's do that, still in this PR, or a follow up? I realised yesterday that we could probably get rid of `forcing` in the TOML altogether, and determine it ourselves, since we fail if a parameters has multiple timesteps but is used statically, and a forcing parameter needs to have that time dimension.

Reply:
My preference is to resolve it in this PR. For the `forcing` part although not required from a code perspective, from a user perspective it may be better that this is explicit (similar to the `cyclic` part). What do you think?

Reply:
Indeed it is nice to be able to look at a TOML file and see which data is forcing data, and which is cyclic. We can make this a separate issue, there is no rush for this.

I can work on h_riv / q_riv early next week. If you have time before, feel free :)

Reply:
Resolved this in commit https://github.com/Deltares/Wflow.jl/pull/128/commits/343b68fed1fa20927e91827a90266e2ab459c1a2, do you agree with these changes?